src/action.js

```diff
@@ -1,52 +1,50 @@
 'use strict'
 
 const core = require('@actions/core')
-const github = require('@actions/github')
-const semverDiff = require('semver/functions/diff')
-const semverCoerce = require('semver/functions/coerce')
 const toolkit = require('actions-toolkit')
 
 const packageInfo = require('../package.json')
 const { githubClient } = require('./github-client')
 const { logInfo, logWarning, logError } = require('./log')
-const {
-  isValidSemver,
-  isCommitHash,
-  getInputs,
-  getPackageName,
-} = require('./util')
-const { targetOptions } = require('./getTargetInput')
-const {
-  getModuleVersionChanges,
-  checkModuleVersionChanges,
-} = require('./moduleVersionChanges')
+const { getInputs, parseCommaOrSemicolonSeparatedValue } = require('./util')
 const { verifyCommits } = require('./verifyCommitSignatures')
 const { dependabotAuthor } = require('./getDependabotDetails')
+const { updateTypes } = require('./mapUpdateType')
+const { updateTypesPriority } = require('./mapUpdateType')
+
+module.exports = async function run({
+  github,
+  context,
+  inputs,
+  dependabotMetadata,
+}) {
+  const { updateType } = dependabotMetadata
+  const dependencyNames = parseCommaOrSemicolonSeparatedValue(
+    dependabotMetadata.dependencyNames
+  )
+
+  const {
+    GITHUB_TOKEN,
+    MERGE_METHOD,
+    EXCLUDE_PKGS,
+    MERGE_COMMENT,
+    APPROVE_ONLY,
+    TARGET,
+    PR_NUMBER,
+  } = getInputs(inputs)
 
-const {
-  GITHUB_TOKEN,
-  MERGE_METHOD,
-  EXCLUDE_PKGS,
-  MERGE_COMMENT,
-  APPROVE_ONLY,
-  TARGET,
-  PR_NUMBER,
-} = getInputs()
-
-module.exports = async function run() {
   try {
     toolkit.logActionRefWarning()
 
-    const { pull_request } = github.context.payload
+    const { pull_request } = context.payload
 
     if (!pull_request && !PR_NUMBER) {
       return logError(
         'This action must be used in the context of a Pull Request or with a Pull Request number'
       )
     }
 
-    const client = githubClient(GITHUB_TOKEN)
-
+    const client = githubClient(github, context, GITHUB_TOKEN)
```

Comment:
I think we can simplify this because based on the docs of the script action, here github is already a preauthenticated rest client. see how they use it here https://github.com/actions/github-script#comment-on-an-issue

Reply:
I'm not suggesting that we ditch our githubClient module altogether, just that we don't need to reauthenticate and create a new rest client